components/core/tests/test-end_to_end.cpp

```diff
@@ -0,0 +1,117 @@
+#include <sys/wait.h>
+
+#include <cstdlib>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp_s/JsonConstructor.hpp"
+#include "../src/clp_s/JsonParser.hpp"
+
+auto const cDefaultTargetEncodedSize = 8ULL * 1024 * 1024 * 1024;  // 8 GB
+auto const cDefaultMaxDocumentSize = 512ULL * 1024 * 1024;  // 512 MB
+auto const cDefaultMinTableSize = 1ULL * 1024 * 1024;  // 1 MB
+auto const cDefaultCompressionLevel = 3;
+auto const cDefaultPrintArchiveStats = false;
+auto const cDefaultOrdered = false;
+auto const cDefaultTargetOrderedChunkSize = 0;
+
+std::string const cTestEndToEndArchiveDirectory{"test-end-to-end-archive"};
+std::string const cTestEndToEndOutputDirectory = "test-end-to-end-out";
+std::string const cTestEndToEndOutputSortedJson = "test-end-to-end_sorted.json";
+std::string const cTestEndToEndInputFileDirectory = "test_log_files";
+std::string const cTestEndToEndInputFile = "test_no_floats_sorted.json";
+
+namespace {
+
+class Cleanup {
+public:
+    Cleanup() {
+        std::filesystem::remove_all(cTestEndToEndArchiveDirectory);
+        std::filesystem::remove_all(cTestEndToEndOutputDirectory);
+        std::filesystem::remove(cTestEndToEndOutputSortedJson);
```

Comment:
Can we move this into a static private method that we call from both the constructor and destructor?